ambry-router/src/main/java/com/github/ambry/router/SimpleOperationTracker.java

```diff
@@ -429,9 +429,10 @@ public boolean maybeFailedDueToOfflineReplicas() {
       logger.info(
           "Terminating {} on {} due to disk down count and total Not_Found count from eligible replicas and some "
               + "other replicas being unavailable. CrossColoEnabled: {}, DiskDownCount: {}, TotalNotFoundCount: {}, "
-              + "TotalReplicaCount: {}, replicaSuccessTarget: {}, OfflineReplicaCount: {}, allReplicaCount: {} {}",
+              + "TotalReplicaCount: {}, replicaSuccessTarget: {}, OfflineReplicaCount: {}, allReplicaCount: {} {} "
+              + "replicaPool = {}",
           routerOperation, partitionId, crossColoEnabled, diskDownCount, totalNotFoundCount, totalReplicaCount,
-          replicaSuccessTarget, totalOfflineReplicaCount, allReplicaCount, getBlobIdLog());
+          replicaSuccessTarget, totalOfflineReplicaCount, allReplicaCount, getBlobIdLog(), replicaPool);
```

Comment:
I am afraid that dumping replicaPool is not enough. Replica Pool only tells us the "good" replicas, we still don't know the "bad" replicas' states.

We should probably construct a map from replica Id to its replica state for all replicas and then dump this map here.

Reply:
You are right. Submit the second revision.